src/main/java/net/neoforged/neoforge/mixins/BlockEntityTypeAccessor.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.mixins;
+
+import java.util.Set;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(BlockEntityType.class)
+public interface BlockEntityTypeAccessor {
+    @Mutable
+    @Accessor
+    void setValidBlocks(Set<Block> validBlocks);
```

Comment:
As far as I'm aware, accessors and invokers are not automatically prefixed for collision avoidance, so this should ideally be named `neoforge$setValidBlocks` and the target field name explicitly specified in the `@Accessor` annotation.